subsys/bluetooth/services/ots/ots.c

```diff
@@ -188,7 +188,7 @@ ssize_t ots_obj_name_write(struct bt_conn *conn,
 	}
 
 	/* Update real object name after no duplicate detected */
-	strcpy(ots->cur_obj->metadata.name, name);
+	utf8_lcpy(ots->cur_obj->metadata.name, name, CONFIG_BT_OTS_OBJ_MAX_NAME_LEN + 1);
```

Comment:
Instead of replacing this `strcpy` with `utf8_lcpy`, perhaps we should just instead use `utf8_trunc` on `name`, or somehow replace the `memcpy` in line 178 with `utf8_lcpy`, as we are already, before this point, using and comparing `name` as a string.

Reply:
Great point, I couldn't use `utf8_lcpy` since `buf` is not null terminated. But you make a great point, we should truncate the name prior to doing the comparison otherwise we may miss duplicates.